runtime/core/exec_aten/util/tensor_util_portable.cpp

```diff
@@ -147,6 +147,11 @@ Error share_tensor_data(
       t_dst.nbytes(),
       t_src.nbytes());
 
+  if (t_src.numel() == 0) {
```

Comment:
Wouldnt you still want to alias the pointers? You can have numel == 0 but not have a null data pointer in the world of dynamic shape.